src/EnergyPlus/SetPointManager.cc

```diff
@@ -7076,45 +6992,30 @@ namespace SetPointManager {
 		Names.allocate( NumVariables );
 
 		GetMeteredVariables( TypeOfComp, NameOfComp, VarIndexes, VarTypes, IndexTypes, UnitsStrings, ResourceTypes, EndUses, Groups, Names, NumFound );
-		IdealCondEntSetPtMgr( SetPtMgrNum ).ChlPumpVarType = VarTypes( 1 );
-		IdealCondEntSetPtMgr( SetPtMgrNum ).ChlPumpVarIndex = VarIndexes( 1 );
-
-		VarIndexes.deallocate();
-		VarTypes.deallocate();
-		IndexTypes.deallocate();
-		UnitsStrings.deallocate();
-		ResourceTypes.deallocate();
-		EndUses.deallocate();
-		Groups.deallocate();
-		Names.deallocate();
-
-		TypeOfComp = PlantLoop( TowerLoopNum ).LoopSide( SupplySide ).Branch( TowerBranchNum ).Comp( TowerNum ).TypeOf;
-		NameOfComp = PlantLoop( TowerLoopNum ).LoopSide( SupplySide ).Branch( TowerBranchNum ).Comp( TowerNum ).Name;
-		NumVariables = GetNumMeteredVariables( TypeOfComp, NameOfComp );
-		VarIndexes.allocate( NumVariables );
-		VarTypes.allocate( NumVariables );
-		IndexTypes.allocate( NumVariables );
-		UnitsStrings.allocate( NumVariables );
-		ResourceTypes.allocate( NumVariables );
-		EndUses.allocate( NumVariables );
-		Groups.allocate( NumVariables );
-		Names.allocate( NumVariables );
-
-		GetMeteredVariables( TypeOfComp, NameOfComp, VarIndexes, VarTypes, IndexTypes, UnitsStrings, ResourceTypes, EndUses, Groups, Names, NumFound );
-		IdealCondEntSetPtMgr( SetPtMgrNum ).ClTowerVarType = VarTypes( 1 );
-		IdealCondEntSetPtMgr( SetPtMgrNum ).ClTowerVarIndex = VarIndexes( 1 );
-
-		VarIndexes.deallocate();
-		VarTypes.deallocate();
-		IndexTypes.deallocate();
-		UnitsStrings.deallocate();
-		ResourceTypes.deallocate();
-		EndUses.deallocate();
-		Groups.deallocate();
-		Names.deallocate();
-
-		TypeOfComp = PlantLoop( CondLoopNum ).LoopSide( SupplySide ).Branch( CondPumpBranchNum ).Comp( CondPumpNum ).TypeOf;
-		NameOfComp = PlantLoop( CondLoopNum ).LoopSide( SupplySide ).Branch( CondPumpBranchNum ).Comp( CondPumpNum ).Name;
+		this->ChlPumpVarType = VarTypes( 1 );
+		this->ChlPumpVarIndex = VarIndexes( 1 );
+
+		for ( int i = 1; i <= this->numTowers; i++ ) {
+			TypeOfComp = PlantLoop( TowerLoopNum ).LoopSide( SupplySide ).Branch( this->CondTowerBranchNum( i ) ).Comp( this->TowerNum( i ) ).TypeOf;
+			NameOfComp = PlantLoop( TowerLoopNum ).LoopSide( SupplySide ).Branch( this->CondTowerBranchNum( i ) ).Comp( this->TowerNum( i ) ).Name;
+			NumVariables = GetNumMeteredVariables( TypeOfComp, NameOfComp );
+			VarIndexes.allocate( NumVariables );
+			VarTypes.allocate( NumVariables );
+			IndexTypes.allocate( NumVariables );
+			UnitsStrings.allocate( NumVariables );
+			ResourceTypes.allocate( NumVariables );
+			EndUses.allocate( NumVariables );
+			Groups.allocate( NumVariables );
+			Names.allocate( NumVariables );
+
+			GetMeteredVariables( TypeOfComp, NameOfComp, VarIndexes, VarTypes, IndexTypes, UnitsStrings, ResourceTypes, EndUses, Groups, Names, NumFound );
+			this->ClTowerVarType.push_back( VarTypes( 1 ) );
+			this->ClTowerVarIndex.push_back( VarIndexes( 1 ) );
+
```

Comment:
This loop gets the var type and var index for multiple towers on the condenser loop.